platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/ConfigurationContextImpl.java

```diff
@@ -147,6 +147,27 @@ public boolean shouldBeVisibleToPlugins() {
     return servicePid != null && configIsNew == null && pidList == null && propertyCount > 0;
   }
 
+  // Contexts can be used in a set
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
✏️  This can be replaced with `if (!getClass().isInstance(o)) {` but usually, one simply checks if `o` is not an instance of the current class (i.e. `ConfigurationContextImpl`) as in `if (!(o instanceof ConfigurationContextImpl)) {`

Reply:
We should still keep the null protection. Changing the rest. 👍

Reply:
`null instanceof <something>` will always be `false`
As for subclasses, it won't break. It will verify the base class is correct which is what you want. The subclass would typically be calling super.equals() and then test for its own instance before verifying its own attributes.

Reply:
I still think using `instanceof` would violate the least astonishment principle. An instance of a different, albeit related, class should not evaluate as equal.